🏗✨：add new ruby-debian library script (#10)
OpenINF/docker-fish: library-scripts/docs/ruby.md

## Ruby Install Script

*Installs Ruby, rbenv.*

**Script status**: Stable

**OS support**: Debian 9+, Ubuntu 18.04+, and downstream distros.

**Maintainer:** The OpenINF Community

### Syntax

```text

> ./ruby-debian.sh [Ruby version] [Non-root user] [Add rvm to rc files flag]

Author: ```suggestion
./ruby-debian.sh [Ruby version] [Non-root user] [Add rbenv to rc files flag]
```

```

Or as a feature:

```json
"features": {
    "ruby": "latest"
}
```

| Argument             | Feature option | Default     | Description                                                                                                                                                                                                                                                                 |
|----------------------|----------------|-------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Ruby version         | `version`      | `latest`    | Version of Ruby to install. A value of `none` will skip this step and just install rvm, rbenv, and tools.                                                                                                                                                                   |

Author: ```suggestion
| Ruby version         | `version`      | `latest`    | Version of Ruby to install. A value of `none` will skip this step and just install rbenv.                                                                                                                                                                   |
```

## Ruby Install Script

> *Installs Ruby, rvm, rbenv, common Ruby utilities, and needed dependencies.*

Author: ```suggestion
*Installs Ruby, rbenv, common Ruby utilities, and needed dependencies.*
```

**Script status**: Stable

**OS support**: Debian 9+, Ubuntu 18.04+, and downstream distros.

> **Maintainer:** The VS Code and GitHub Codespaces teams

Author: ```suggestion
**Maintainer:** The OpenINF Community
```

|----------------------|----------------|-------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Ruby version         | `version`      | `latest`    | Version of Ruby to install. A value of `none` will skip this step and just install rvm, rbenv, and tools.                                                                                                                                                                   |
| Non-root user        |                | `automatic` | Specifies a user in the container other than root that will use Ruby, rvm, and/or rbenv. A value of `automatic` will cause the script to check for a user called `vscode`, then `node`, `codespace`, and finally a user with a UID of `1000` before falling back to `root`. |

> | Add to rc files flag |                | `true`      | A `true`/`false` flag that indicates whether sourcing the rvm script should be added to `/etc/bash.bashrc` and `/etc/zsh/zshrc`.                                                                                                                                            |

Author: ```suggestion
| Add to rc files flag |                | `true`      | A `true`/`false` flag that indicates whether sourcing the rvm script should be added to `/etc/fish/config.fish`.                                                                                                                                            |
```

| Install tools flag   |                | `true`      | A `true`/`false` flag that indicates whether tools like `rake`, `ruby-debug-ide`, and `debase` should be installed.                                                                                                                                                         |

### Usage

> #### Feature use

Author: ```suggestion
#### Feature Use
```

To install these capabilities in your primary dev container, reference it in `devcontainer.json` as follows:

```json
"features": {
    "ruby": "latest"
}
```

If you have already built your development container, run the **Rebuild Container** command from the command palette (<kbd>Ctrl/Cmd</kbd> + <kbd>Shift</kbd> + <kbd>P</kbd> or <kbd>F1</kbd>) to pick up the change.

> #### Script use

Author: ```suggestion
#### Script Use
```
